### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**
[DEMO LINK](https://vadym4che.github.io/react_movies-list-filter/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -20,12 +48,14 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={(event) => handleSearch(event)}
```
Comment (Andr1yK) [inline src/App.tsx:52]:
```suggestion
                onChange={handleSearch}
```

**Review by Andr1yK — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+    });
+
+    setMoviesToShow(filtered);
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
+      filterMovies();
+    }, 500);
+  }, [query]);
```
[the remark this answers is not in the document]
Comment (vadym4che) [reply src/App.tsx:33]: I apologize, - I was just playing with script and have forgotten to remove this childishness

File: src/App.tsx (hunk 1 above)
Comment (vadym4che) [reply src/App.tsx:52]: unwrapped

File: src/App.tsx [hunk 3]
```diff
@@ -1,9 +1,37 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [moviesToShow, setMoviesToShow] = useState(moviesFromServer);
```
[the remark this answers is not in the document]
Comment (vadym4che) [reply src/App.tsx:8]: done - now rerendering happens only when this.state.query has been changed

**Review by Andr1yK — APPROVED** [review]

**Pull request closed without merging**